### ISSUE TITLE
Olympus Application Info

### ISSUE THREAD

**Issue by tazamyinsight** [Azure/Azure-Sentinel#11605]
Hello everyone, I have been seeing non-interactive sign-in logs in Entra ID, where the application is called "Olympus" and the resource is called "OCaaS Client Interaction Service". I see OCaaS is a Microsoft service. Can someone please tell me any information regarding Olympus and its correlation to OCaaS/Microsoft?

**Comment by tazamyinsight** [comment]
I found this Github post about it but not sure what Olympus application does/what its used for https://github.com/merill/microsoft-info/commit/0baa94b0e8a2d729cf72c18c4b056672db1c3ef6

**Comment by v-sudkharat** [comment]
@tazamyinsight, 
Sharing a blog link which help you to understand the use of Olympus : 

https://learn.microsoft.com/en-us/answers/questions/793759/conditional-access-policy-that-blocks-sign-ins-fro

![Image](https://github.com/user-attachments/assets/1393c3cc-2a62-4f13-afe9-5cc44e7cba9a)

Thanks!

**Comment by v-sudkharat** [comment]
@tazamyinsight, 
Hope above shared blog helps, kindly let us know, can we close this issue?

**Comment by v-sudkharat** [comment]
Hi @tazamyinsight, Actually, Olympus is likely an internal Microsoft service, involved in handling non-interactive sign-ins or background flows related to Office apps (Ex. Outlook, OneDrive) and other Microsoft services. and “Olympus” application is not as documented as **OCaaS**, but it is likely an internal or specialized application used within Microsoft. 
If you need more specific details about “Olympus,” it might be helpful to reach out to our Microsoft support: https://support.microsoft.com/en-us/contactus

Thanks!

**Comment by v-sudkharat** [comment]
@tazamyinsight , closing this issue from GitHub. If you still need support for this issue, feel free to re-open it any time. Thank you for your co-operation.

**Comment by mdavistffhrtporg** [comment]
Hi @v-sudkharat I'm finding this in my Entra ID logs also. I would like to request that you please re-open this issue and provide some additional documentation on Olympus. For reference, we do us Sentinel.

**Comment by andres-swax** [comment]
+1 here. Showing on our AAD logs.

Further info: All logins are the same type: non-interactive and taking place at times which the user can be at the computer or not so no distinguishing factor there. Just two users are experiencing it one of those is myself in fact.

On the Device Info section the device ID points to a valid computer (for that user) and the Browser shows `Rich Client 4.69.1.0`.

A quick web search identifies this as a .NET library called `Microsoft.Identity.Client` part of the Identity Platform. So, that's _how_ the call gets here but not _by whom_ or _why_.

Resource it is trying to access is `OCaaS Client Interaction Service` which is an enterprise application.

Everyone at the office has Onedrive logged into a business account and from there to the Sign-in Logs for this enterprise application, there are no signins interactive, non- or otherwise, meaning _the _application_ ot tried to login using its user principal, but the users have tried to access the application (incoming, not outgoing).

If I go to `Enterprise Applications` in the Entra portal

I'm filing this in the Annoyingly Misterious bin but I feel it's 'ok' if thats an appropiate qualifier.

This post seems to (have become to) be ground-zero for matter. Nowhere else on my web searches I found anything related.

**Comment by RickKotlarz** [comment]
Based on comments from AmanpreetSingh-MSFT found here [https://learn.microsoft.com/en-us/answers/questions/793759/conditional-access-policy-that-blocks-sign-ins-fro](https://learn.microsoft.com/en-us/answers/questions/793759/conditional-access-policy-that-blocks-sign-ins-fro) and the list of 1st party applications found here [https://github.com/merill/microsoft-info/?tab=readme-ov-file](https://github.com/merill/microsoft-info/?tab=readme-ov-file) the app with AppId `c2ada927-a9e2-4564-aae2-70775a2fa0af` and AppDisplayName "OCaaS Client Interaction Service" is a Microsoft service principal associated with Microsoft Entra ID.

It is a default service principal present in all Azure AD tenants and is primarily accessed by Office client applications such as Outlook and OneDrive to complete authentication and authorization tasks.

This service principal is often involved in scenarios where conditional access policies are evaluated, especially in "bootstrap" scenarios. In such cases, the evaluation may be skipped to ensure that essential services like authentication are not blocked.

The AppOwnerOrganizationId `f8cdef31-a31e-4b4a-93e4-5f571e91255a` indicates that this service principal is owned by Microsoft, and the "Source" field labeled "Graph" suggests that it is managed through Microsoft Graph API.

In summary, the "OCaaS Client Interaction Service" is a Microsoft-managed service principal used to facilitate authentication and authorization processes for Office client applications within Azure AD tenants.
